src/util/dataReducers.js

```diff
@@ -432,3 +432,211 @@ export const getDecodedDFTable = (dfResult) =>{
 }
   return {data: result, headers: keys};
 };
+
+export const getFeatureNames = (dfResult) =>{
+  let keys = Object.keys(dfResult);
+  return keys;
+};
+
+export const getUniqueFeatureValues = (dfResult) =>{
+  let i=0; let j=0;
+  let encodedResult = {};
+  let decodedResult = {};
+  let keys = Object.keys(dfResult);
+  if (keys.length>0) {
+    for (i = 0; i<keys.length; i++) {
+      let arr = [];
+      let key = keys[i];
+      let decodedKeys = Object.keys(dfResult[key]);
+      for (j = 0; j < decodedKeys.length; j++) {
+        arr.push(dfResult[key][decodedKeys[j]]);
+    }
+    encodedResult[key] = arr;
+    decodedResult[key] = decodedKeys;
+    }
+  }
+
+  return {'encodedResult': encodedResult, 'decodedResult': decodedResult};
+};
+
+export const parseCfFeedbackResult = (result) =>{
+  if (JSON.stringify(result) != '{}') {
+    let keys = Object.keys(result['freq_seqs_after_filter']);
+    if (keys.length>0) {
+      return {'tp': result['freq_seqs_after_filter']['tp'], 'tn': result['freq_seqs_after_filter']['tn'],
+        'fp': result['freq_seqs_after_filter']['fp'], 'fn': result['freq_seqs_after_filter']['fn']};
+    }
+  }
+  return {'tp': '', 'tn': '', 'fp': '', 'fn': ''};
+};
+
+export const getUniquePatterns = (result) =>{
+  let resultArray = new Set();
+  let i = 0;
+  let j = 0;
+  let k = 0;
+  if (result == null) return [];
+  else {
+    let keys = Object.keys(result['freq_seqs_after_filter']);
+    for (i=0; i<keys.length; i++) {
+      if (result['freq_seqs_after_filter'][keys[i]].length>0 ) {
+        for (j=0; j<result['freq_seqs_after_filter'][keys[i]].length; j++) {
+          let patterns = '';
+          for (k=0; k<result['freq_seqs_after_filter'][keys[i]][j][0].length; k++) {
+            if (k < result['freq_seqs_after_filter'][keys[i]][j][0].length-1) {
+              patterns+=result['freq_seqs_after_filter'][keys[i]][j][0][k]+'; ';
+            } else {
+              patterns+=result['freq_seqs_after_filter'][keys[i]][j][0][k];
+            }
+          }
+          resultArray = resultArray.add(patterns);
+        }
+      }
+    }
+  }
+  return Array.from(resultArray);
+};
+
+export const getPatternsForMatrix = (result, matrixName) =>{
+  let resultArray = [];
+  let i = 0;
+  let j = 0;
+  let k = 0;
+  if (JSON.stringify(result) != '{}') {
+    const uniquePatterns = encodePatternsForDropdown(result);
+    if (matrixName == 'All') {
+      let keys = Object.keys(result['freq_seqs_after_filter']);
+      for (i=0; i<keys.length; i++) {
+        if (result['freq_seqs_after_filter'][keys[i]].length>0 ) {
+          for (j=0; j<result['freq_seqs_after_filter'][keys[i]].length; j++) {
+            let arr = [];
+            let patterns = '';
+            arr.push(getConfusionMatrixName(keys[i]));
+            for (k=0; k<result['freq_seqs_after_filter'][keys[i]][j][0].length; k++) {
```

Comment:
please be coherent when you write assignments, like k=0, or comparisons, like k<result, in the amount of spaced used (see https://standardjs.com/rules-en.html infix operators)